README.md

```diff
@@ -46,70 +38,172 @@ Maven:
+You can encrypt messages on the sending end in the following way:
+
+```java
+final Box box = context.encrypt(yourData);
+dataChannel.send(ByteBuffer.wrap(box.toBytes()));
+```
+
+On the receiving end, decrypt the message by the use of the crypto context:
+
+```java
+final Box box = new Box(message);
+final byte[] yourData = context.decrypt(
+    box, DataChannelCryptoContext.NONCE_LENGTH);
+```
+
+Note, that you should not use a crypto context for a data channel that is being
+used for handover. The task will take care of encryption and decryption itself.
+
+### Handover
 
-Once the signaling channel is open, create offer/answer/candidates as usual and
-send them through the signaling channel using the corresponding methods:
+Before initiating the handover, the application needs to fetch the
+`ISignalingTransportLink` instance which contains the necessary information to
+create a data channel.
 
-- `task.sendOffer(offer)`
-- `task.sendAnswer(answer)`
-- `task.sendCandidates(candidates)`
+```java
+final ISignalingTransportLink link = task.getTransportLink();
 
-As soon as the data channel is open, request a handover of the signaling channel:
+final DataChannel.Init parameters = new DataChannel.Init();
+parameters.id = link.getId();
+parameters.negotiated = true;
+parameters.ordered = true;
+parameters.protocol = link.getProtocol();
+
+final DataChannel dataChannel = peerConnection.createDataChannel(
+    link.getLabel(), parameters);
+```
+
+Note that the data channel used for handover **must** be created with the
+label and parameters as shown in the above code snippet.
+
+Now that you have created the channel, you need to implement the
+`ISignalingTransportHandler` interface. Below is a minimal handler that forwards
+the necessary events and messages to the created data channel.
 
 ```java
-dc.registerObserver(new DataChannel.Observer() {
-    // ...
+final ISignalingTransportHandler handler = new ISignalingTransportHandler() {
+    @Override
+    public long getMaxMessageSize() {
+        return peerConnection.sctp().getMaxMessageSize();
+    }
+
+    @Override
+    public void close() {
+        dataChannel.close();
+    }
+
+    @Override
+    public void send(@NonNull final ByteBuffer message) {
+        // Note: Always send binary
+        dataChannel.send(new DataChannel.Buffer(message, true));
+    }
+};
+```
+
+Furthermore, you have to bind all necessary events in order to connect the data
+channel to the `ISignalingTransportLink`.
+
+```java
+dataChannel.registerObserver(new DataChannel.Observer() {
+    // [...]
 
     @Override
     public void onStateChange() {
-        if (dc.state() == DataChannel.State.OPEN) {
-            task.handover();
+        switch (dataChannel.state()) {
+            case OPEN:
+                task.handover(handler);
+                break;
+            case CLOSING:
+                link.closing();
+                break;
+            case CLOSED:
+                link.closed();
+                break;
```

Comment:
Example does not handle `UntiedException` in closing() and closed().

Reply:
Mh, true, but do you feel it necessary to add this to the readme? It's not a `RuntimeException`, so I don't mind not catching it in the example.

Reply:
`UntiedException` extends `Exception` and is thus a checked exception which must be caught.

Subtypes of `Error` and `RuntimeException` are unchecked, all other subtypes of `Throwable` are checked.

Reply:
Yeah, I get it, but does it matter for you to understand the example?

Reply:
I think it would be nice if examples would act as "best practice" and could actually be copy-pasted. Otherwise someone getting started with the codebase will be confused by the `UntiedException` and will not know how to handle it without diving into the codebase.

Reply:
It would just clutter the example. There are more places where you'd have to catch that.

Reply:
Not terribly user-friendly for people new to this library (since there *are* no docs besides the README), but that's not our main concern right now, so fair enough.